assets/archive-functions.cmake

```diff
@@ -0,0 +1,27 @@
+function(zip_archive ARCHIVE)
+  # any additional arguments will be passed to the zip utility (see ${ARGN})
+  add_custom_command(
+    OUTPUT "${ARCHIVE}"
+    DEPENDS assets # is a target, not a file, so no issue with relative paths
+    COMMAND "${CMAKE_COMMAND}" -E tar "cfv" "${ARCHIVE}" --format=zip ${ARGN}
```

Comment:
This will fail on some OSes. Why not just use Info-ZIP’s `zip` and `unzip` commands, which, while an extra package, are pretty standard.

Reply:
Why will this fail on some OSes? Surely this works on every OS that CMake runs on?

Reply:
@Jojo-Schmitz this doesn't call GNU tar, it calls CMake's built-in tar command (which exists precisely to avoid platform issues). See [this answer on StackOverflow](https://stackoverflow.com/a/7052049).

Reply:
Ah!, Thanks, wasn't aware of that.

Reply:
Ah, same here, thanks!